src/plugins/data/common/search/es_search/types.ts

```diff
@@ -51,8 +52,37 @@ export interface IEsSearchResponse<Source = any> extends IKibanaSearchResponse {
    * Indicates whether the results returned are complete or partial
    */
   isPartial?: boolean;
-  rawResponse: SearchResponse<Source>;
+  rawResponse: SearchResponse<T>;
```

Comment:
The only response arguments the `search` infrastructure cares about are:
 - `id` - the async search ID provided by elasticsearch
 - `isRunning` and `isPartial` which are indicators of the search's execution in ES.
 - `took` and `total_hits` for telemetry and potentially any additional meta data about the request's execution.

Otherwise, the response can be completely generic, so we could create a new type for `rawResponse` (EsSearch response is specific to the ES strategy).